lib/consent/permissions.rb

```diff
@@ -0,0 +1,37 @@
+module Consent
+  class Permissions
+    include Enumerable
+
+    def initialize(permissions)
+      @permissions = permissions
+    end
+
+    def each(&block)
+      Consent.subjects.values.each do |subject|
+        subject.actions.map do |action|
+          map_permission subject, action
+        end.compact.each(&block)
```

Comment:
would it be possible to write this as `subject.actions.map { |action| map_permission(subject, action) }.compact.each(&block)`  ?

I know it's a slightly long line, but this way you avoid the extra nesting here